python/tvm/relay/op/contrib/tensorrt.py

```diff
@@ -723,8 +723,9 @@ def pad_annotate_fn(expr):  # pylint: disable=unused-variable
     if attrs.pad_mode != "constant":
         logger.info("nn.pad: pad mode is %s but must be constant.", attrs.pad_mode)
         return False
-    if float(attrs.pad_value) != 0.0:
-        logger.info("nn.pad: pad value is %f but must be 0.0.", float(attrs.pad_value))
+    assert isinstance(args[1], relay.Constant)
+    if len(args[1].checked_type.shape) == 0 and args[1].data.numpy().item() != 0.0:
```

Comment:
I think it's better to check type of pad_value with if instead of assert because it may be Expr.

I suggest the following code, including the fix of checking the shape of pad_value,

```suggestion
    if (not isinstance(args[1], relay.Constant) or
        len(args[1].checked_type.shape) != 0 or
        args[1].data.numpy().item() != 0.0):
```

Reply:
It shouldn't be Expr anyways if the pad mode is constant.

Reply:
I think pad_mode="constant" just means padding by pad_value, not type of pad_value is relay.Constant.

nn.pad allows Expr for pad_value.

https://github.com/apache/tvm/blob/03c8a6f4612a91cdbb668482309de33c19ec5949/python/tvm/relay/op/nn/nn.py#L1657-L1658

Reply:
Hmm you're right. I didn't notice that constant mode doesn't enforce constant node. @akmaru please change accordingly.

Reply:
Fixed by https://github.com/apache/tvm/pull/8131/commits/e50bc434b6dd952f843f71192a11edc923f57748, please review again.